src/syscheckd/whodata/syscheck_audit.c

```diff
@@ -186,8 +186,10 @@ void audit_no_rules_to_realtime() {
     int found;
     char *real_path = NULL;
     directory_t *dir_it = NULL;
+    OSListNode *node_it;
 
-    foreach_array(dir_it, syscheck.directories) {
+    OSList_foreach(node_it, syscheck.directories) {
```

Comment:
I know this function is called by the main thread, but we should still lock the configuration before accessing to prevent possible data races in the future.